File: exersice-solutions/cracking-the-coding-interview/cpp/chapter-1/src/urlify.cpp
#include "urlify.h"

#include<string>
#include <regex>

std::string Urlify::urlifyString(std::string s)
{
	s = std::regex_replace(s, std::regex("^ +| +$|( ) +"), "$1");
	s = std::regex_replace(s, std::regex(" "), "%20");
	return s;
}

File: exersice-solutions/cracking-the-coding-interview/CSharp/chapter-1/Chapter1/URLify.cs
using System;
using System.Text.RegularExpressions;

namespace Chapter1
{
    public static class URLify
    {
        public static string URLifyString(string s, int length)
        {
            s = s.Trim();
            s = Regex.Replace(s, @" ", "%20");
            return s;
        }
    }
}
